spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/BoxControllerHelperService.java

```diff
@@ -621,62 +617,34 @@ public JSONObject getBoxBarcode(HttpSession session, JSONObject json) {
-        serviceName = json.getString("serviceName");
+      @Override
+      public Box fetch(long id) throws IOException {
+        return requestManager.getBoxById(id);
       }
 
-      MisoPrintService<File, Barcodable, PrintContext<File>> mps = null;
-      if (serviceName == null) {
-        Collection<MisoPrintService> services = printManager.listPrintServicesByBarcodeableClass(Box.class);
-        if (services.size() == 1) {
-          mps = services.iterator().next();
-          if (mps == null) {
-            return JSONUtils
-                .SimpleJSONError("Unable to resolve a print service for Boxes. A service seems to be recognised but cannot be resolved.");
-          }
-        } else {
-          return JSONUtils
-              .SimpleJSONError("No serviceName specified, but more than one available service able to print this barcode type.");
-        }
-      } else {
-        mps = printManager.getPrintService(serviceName);
-        if (mps == null) {
-          return JSONUtils.SimpleJSONError("Unable to resolve a print service for Boxes with the name '" + serviceName + "'.");
-        }
+      @Override
+      public void store(Box item, User user) throws IOException {
+        item.setLastModifier(user);
+        requestManager.saveBox(item);
       }
 
-      Queue<File> thingsToPrint = new LinkedList<>();
-      JSONArray ss = JSONArray.fromObject(json.getString("boxes"));
-      for (JSONObject s : (Iterable<JSONObject>) ss) {
-        try {
-          Long boxId = s.getLong("boxId");
-          Box box = requestManager.getBoxById(boxId);
-          // autosave the barcode if none has been previously generated
-          if (box.getIdentificationBarcode() == null || "".equals(box.getIdentificationBarcode())) {
-            box.setLastModifier(authorizationManager.getCurrentUser());
-            requestManager.saveBox(box);
-          }
-          File f = mps.getLabelFor(box);
-          if (f != null) thingsToPrint.add(f);
-        } catch (IOException e) {
-          log.debug("Failed to print barcodes", e);
-          return JSONUtils.SimpleJSONError("Error printing barcodes: " + e.getMessage());
-        }
+      @Override
+      public String groupName() {
+        return "boxes";
       }
 
-      PrintJob pj = printManager.print(thingsToPrint, mps.getName(), user);
-      return JSONUtils.SimpleJSONResponse("Job " + pj.getId() + " : Barcodes printed.");
-    } catch (MisoPrintException e) {
-      log.debug("Failed to print barcodes", e);
-      return JSONUtils.SimpleJSONError("Failed to print barcodes: " + e.getMessage());
-    } catch (IOException e) {
-      log.debug("Failed to print barcodes", e);
-      return JSONUtils.SimpleJSONError("Failed to print barcodes: " + e.getMessage());
-    }
+      @Override
+      public String idName() {
+        return "boxId";
+      }
+
+      @Override
+      public Iterable<Box> fetchAll(long projectId) throws IOException {
+        return Collections.emptySet();
```

Comment:
Doc method in interface as optional and throw UnsupportedOperationException here